prevent multiple load balancer groups to match

aryszka opened the pull request:
when a previous load balancer group was already matched, the LBGroup() predicate will always return false

aryszka commented:
The original problem with the Traffic control was hard to test so that it always covers the underlying problem, so the test is now simplified to use groups without members. This way it is always reproduce the difference between not matching the wrong route, represented by a 404, vs ending up in an infinite loopback.

mikkeloscar commented on loadbalancer/predicate.go:
```diff
@@ -39,7 +39,7 @@ func getGroupDecision(h http.Header, group string) (string, bool) {
 	}
 
-	return "", false
+	return "", len(h[DecisionHeader]) > 0
```
nit: I think a doc string for this function would be helpful to understand what is really going on here.

mikkeloscar commented on proxy/proxytest/proxytest.go:
```diff
@@ -42,6 +43,7 @@ func newTestProxy(fr filters.Registry, routingOptions routing.Options, proxyPara
 		routingOptions.Predicates = []routing.PredicateSpec{
 			loadbalancer.NewGroup(),
 			loadbalancer.NewMember(),
+			traffic.New(),
```
Given your last comment, is this needed still? I don't fully understand what it's used for.

aryszka commented:
thanks @mikkeloscar 
removed the unused test changes. For the getGroupDecision() function, instead of doc string, I split the function into two with single responsibility and descriptive names.

mikkeloscar commented on loadbalancer/predicate.go:
```diff
@@ -27,19 +27,23 @@ type memberPredicate struct {
+	return ""
+}
+
+func hasGroupDecision(h http.Header, group string) bool {
```
`group` arg is not used.

mikkeloscar commented:
@aryszka thanks for digging into this!

szuecs commented:
@mikkeloscar we also tested it in playground, that it fixes the problem. I could not stop the investigation. O:)
@aryszka thanks for helping with investigation, opening the PR and writing the tests, awesome to see you always write tests!

szuecs commented:
fixed comment @mikkeloscar

szuecs commented:
:+1:

ese commented:
:100: We were facing some issues due this
